src/sentry/integrations/slack/integration.py

```diff
@@ -157,3 +165,159 @@ def build_integration(self, state):
+
+        return render_to_response(
+            template="sentry/integrations/slack-reauth-details.html",
+            context={
+                "next_url": "%s%s" % (absolute_uri("/extensions/slack/setup/"), next_url_param),
+                "private": channels["private"],
+                "no_permission": channels["no_permission"],
+                "not_found": channels["channel_not_found"],
+            },
+            request=request,
+        )
+
+
+def _request_channel_info(pipeline):
+    channels = pipeline.fetch_state("all_channels")
+    integration_id = pipeline.fetch_state("integration_id")
+
+    try:
+        integration = Integration.objects.get(id=integration_id, status=0, provider="slack",)
+    except Integration.DoesNotExist:
+        # probably raise an IntegrationError here
+        return
+
+    channel_responses = defaultdict(lambda: set())
+    for channel in channels:
+        payload = {
+            "token": integration.metadata["access_token"],
+            "channel": channel["id"],
+        }
+        client = SlackClient()
+        try:
+            resp = client.post("/conversations.info", data=payload)
+        except ApiError as e:
+            # adds the channel to our dict grouped by the error message which could
+            # be any of the following found under the 'errors' section found in
+            # https://api.slack.com/methods/conversations.list
+
+            # we would be catching other errors too that we don't care about
+            # but maybe that's fine to just add em in
+            channel_responses[e].add(Channel(channel["name"], channel["id"]))
+            continue
+
+        if resp["channel"]["is_private"]:
+            channel_responses["private"].add(Channel(channel["name"], channel["id"]))
+
+    pipeline.bind_state("private_channels", channel_responses["private"])
+    return channel_responses
+
+
+def _get_channels_from_rules(pipeline):
+    organization = pipeline.organization
+    integration_id = pipeline.fetch_state("integration_id")
+
+    try:
+        integration = Integration.objects.get(id=integration_id, provider="slack",)
+    except Integration.DoesNotExist:
+        # probably raise an IntegrationError here
+        return
+
+    rules = Rule.objects.filter(
+        project__in=organization.project_set.all(), status=RuleStatus.ACTIVE,
+    )
+
+    channels = []
+    for rule in rules:
+        # try and see if its used for slack
+        for rule_action in rule.data["actions"]:
+            rule_integration_id = rule_action.get("workspace")
+            if rule_integration_id and rule_integration_id == six.text_type(integration.id):
+
+                channel_id = rule_action["channel_id"]
+                channel_name = rule_action["channel"]
+
+                # don't care if its a user
+                if channel_name[0] == "@":
+                    continue
+
+                channels.append({"name": channel_name, "id": channel_id})
```

Comment:
@MeredithAnya This function needs to de-dupe the channels since you could have multiple alert rules going to the same channel